src/search/engines/bing/bing.go

```diff
@@ -70,28 +60,39 @@ func Search(ctx context.Context, query string, relay *bucket.Relay, options engi
 		}
 	})
 
-	localeParam := getLocale(&options)
+	localeParam := getLocale(options)
 
+	// create a buffered channel to hold errors, size of channel is the number of pages = requests being made
+	errChannel := make(chan error, options.MaxPages)
 	colCtx := colly.NewContext()
 	colCtx.Put("page", strconv.Itoa(1))
 
 	urll := Info.URL + query + localeParam
 	anonUrll := Info.URL + anonymize.String(query) + localeParam
-	_sedefaults.DoGetRequest(urll, anonUrll, colCtx, col, Info.Name, &retError)
+	_sedefaults.DoGetRequest(urll, anonUrll, colCtx, col, Info.Name, errChannel)
 
 	for i := 1; i < options.MaxPages; i++ {
 		colCtx = colly.NewContext()
 		colCtx.Put("page", strconv.Itoa(i+1))
 
 		urll := Info.URL + query + "&first=" + strconv.Itoa(i*10+1) + localeParam
 		anonUrll := Info.URL + anonymize.String(query) + "&first=" + strconv.Itoa(i*10+1) + localeParam
-		_sedefaults.DoGetRequest(urll, anonUrll, colCtx, col, Info.Name, &retError)
+		_sedefaults.DoGetRequest(urll, anonUrll, colCtx, col, Info.Name, errChannel)
+	}
+
+	// only store errors that occur and are not Timeout errors
+	retErrors := make([]error, 0)
+	for i := 0; i < options.MaxPages; i++ {
+		err := <-errChannel
+		if err != nil && !engines.IsTimeoutError(err) {
+			retErrors = append(retErrors, err)
+		}
```

Comment:
Why does this care if the Error is a Timeout and other engines dont? Can DoGetRequest even return a Timeout Error?